src/Tuura/Plato/Translation.hs

```diff
@@ -1,17 +1,25 @@
 module Tuura.Plato.Translation where
 
 import Data.Char
-import Data.List.Extra
 
 import Tuura.Concept.Circuit.Basic
 import Tuura.Concept.Circuit.Derived
 
 data ValidationResult a = Valid | Invalid [ValidationError a] deriving Eq
 
+instance Monoid (ValidationResult a) where
+    mempty = mempty
+
+    mappend Valid (Invalid es) = Invalid es
+    mappend (Invalid es) Valid = Invalid es
+    mappend Valid Valid = Valid
+    mappend (Invalid es) (Invalid fs) = Invalid (fs ++ es)
+    -- mappend (Defined x) (Defined y) = if x == y then Defined x else Inconsistent
+
 data ValidationError a = UnusedSignal a
                        | InconsistentInitialState a
                        | UndefinedInitialState a
-                       | InvariantViolated a
+                       | InvariantViolated [Transition a]
```

Comment:
@jrbeaumont Sure, let's come back to this. In fact, you can postpone this change until a later PR if that makes things easier. (I'm pretty sure though that the proposed change will eventually work.)